lib/src/boundimage.rs

```diff
@@ -0,0 +1,171 @@
+use crate::task::Task;
+use anyhow::{Context, Result};
+use camino::Utf8Path;
+use fn_error_context::context;
+use ostree_ext::ostree::Deployment;
+use ostree_ext::sysroot::SysrootLock;
+use regex::Regex;
+use rustix::fs::{OFlags, ResolveFlags};
+use std::fs;
+use std::fs::File;
+use std::io::Read;
+use std::os::unix::io::AsFd;
+use std::path::Path;
+
+const BOUND_IMAGE_DIR: &'static str = "usr/lib/bootc/bound-images.d";
+
+pub(crate) struct BoundImageManager {
```

Comment:
I'm generally OK with this, however...AFAICS we basically create the struct to call a single method on it (`run`) right?

Do we need the struct at all then? In Rust (unlike e.g. [Java which forces everything in a class](http://steve-yegge.blogspot.com/2006/03/execution-in-kingdom-of-nouns.html) or Python which inherited some of the OOP (but also supports free functions)), Rust has a lot more of the sense of "modules are namespaces, and free functions are totally OK".

For example the [std::fs module](https://doc.rust-lang.org/std/fs/index.html) has a ton of pure free functions.

In our own [kargs.rs](https://github.com/containers/bootc/blob/main/lib/src/kargs.rs) - it's just free functions, and I think personally that's fine for what it does.

In this case I think it'd be equally clear if the sole "external" entrypoint was something like

```
pub(crate) fn pull_bound_images(sysroot: &SysrootLock, deployment: &Deployment) -> Result<()>
```
or so, right?

The unit of "encapsulation" here is a module, not a struct; the default `fn` is module private.

But again, I don't have a *really* strong opinion here, this is mainly a style thing.